site/app/controllers/grading/ElectronicGraderController.php

```diff
@@ -132,65 +143,79 @@ public function submitGrade() {
         $who_id = $_POST['u_id'];
         $gradeable = $this->core->getQueries()->getGradeable($gradeable_id, $who_id);
 
+        if ($this->core->getUser()->getGroup() > $gradeable->getMinimumGradingGroup()) {
+            $_SESSION['messages']['error'][] = "You do not have permission to grade {$gradeable->getName()}";
+            $this->core->redirect($this->core->getConfig()->getSiteUrl());
+        }
+
         $now = new \DateTime('now', $this->core->getConfig()->getTimezone());
         $homeworkDate = $gradeable->getGradeStartDate();
         if ($now < $homeworkDate) {
             $_SESSION['messages']['error'][] = "Grading is not open yet for {$gradeable->getName()}";
             $this->core->redirect($this->core->buildUrl(array('component'=>'grading', 'page'=>'electronic', 'gradeable_id' => $gradeable_id)));
         }
+
+        if (isset($_POST['overwrite'])) $gradeable->setGrader($this->core->getUser());
```

Comment:
And we save that on line 164 for each component, which is what we're using to track who was the grader, not saving a grader for the whole gradeable.